source/base/mpi_compute_index_owner_internal.cc

```diff
@@ -437,14 +439,13 @@ namespace Utilities
 
           Assert(size > 0, ExcNotImplemented());
 
-          dofs_per_process = (size + n_procs - 1) / n_procs;
-          if (dofs_per_process < range_minimum_grain_size)
-            {
-              dofs_per_process  = range_minimum_grain_size;
-              stride_small_size = dofs_per_process * n_procs / size;
-            }
-          else
-            stride_small_size = 1;
+          dofs_per_process =
+            std::max<types::global_dof_index>((size + n_procs - 1) / n_procs,
+                                              range_minimum_grain_size);
```

Comment:
Is this right? I think you only want to set `stride_small_size`, but not modify `dofs_per_process`.

Reply:
I think this should be ok, because we previously set `dofs_per_process` and modified it if it was below the threshold.

Reply:
After outsourcing the `stride_small_size = std::max(...)` assignment in commit 034fee0c8bcf7a5b33683a97745c8b34eff22f7c, the remaining lines regarding `dofs_per_process` were:
```
dofs_per_process = (size + n_procs - 1) / n_procs;
if (dofs_per_process < range_minimum_grain_size)
  dofs_per_process = range_minimum_grain_size;
```
This looked like a good application for `std::max` to me.

---

We can debate about whether my commit in 034fee0c8bcf7a5b33683a97745c8b34eff22f7c is right in the first place.